platform/linux-generic/arch/aarch64/odp/api/abi/hash.h

```diff
@@ -0,0 +1,67 @@
+/* Copyright (c) 2021 ARM Limited
+ * All rights reserved.
+ *
+ * SPDX-License-Identifier:     BSD-3-Clause
+ */
+
+#ifndef ODP_API_ABI_HASH_H_
+#define ODP_API_ABI_HASH_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#ifdef __ARM_FEATURE_CRC32
+#define _ODP_CRC32_EXTENSIONS
+
+#include <arm_acle.h>
+
+static inline uint32_t crc32_u8(uint8_t data, uint32_t init_val)
+{
+	return __crc32b(init_val, data);
+}
+
+static inline uint32_t crc32_u16(uint16_t data, uint32_t init_val)
+{
+	return __crc32h(init_val, data);
+}
+
+static inline uint32_t crc32_u32(uint32_t data, uint32_t init_val)
+{
+	return __crc32w(init_val, data);
+}
+
+static inline uint64_t crc32_u64(uint64_t data, uint32_t init_val)
+{
+	return __crc32d(init_val, data);
+}
+
+static inline uint32_t crc32c_u8(uint8_t data, uint32_t init_val)
+{
+	return __crc32cb(init_val, data);
+}
+
+static inline uint32_t crc32c_u16(uint16_t data, uint32_t init_val)
+{
+	return __crc32ch(init_val, data);
+}
+
+static inline uint32_t crc32c_u32(uint32_t data, uint32_t init_val)
+{
+	return __crc32cw(init_val, data);
+}
+
+static inline uint64_t crc32c_u64(uint64_t data, uint32_t init_val)
+{
+	return __crc32cd(init_val, data);
+}
+
+#endif
+
+#ifdef __cplusplus
+}
+#endif
+
+#include <odp/api/abi-default/hash.h>
```

Comment:
This should not be included here, I think. Only the empty architecture-specific headers include the abi-default header.

Reply:
Okay, I will remove <odp/api/abi-default/hash.h>. Thanks!